src/main/java/com/google/api/codegen/transformer/go/GoSurfaceNamer.java

```diff
@@ -0,0 +1,198 @@
+      return super.getDocLines(element);
+    }
+    Method method = (Method) element;
+    String text = DocumentationUtil.getDescription(method);
+    text = lowerFirstLetter(text);
+    return super.getDocLines(getApiMethodName(method) + " " + text);
+  }
+
+  @Override
+  public String getAndSavePagedResponseTypeName(ModelTypeTable typeTable, TypeRef resourceType) {
+    String typeName = converter.getTypeNameForElementType(resourceType).getNickname();
+    int dotIndex = typeName.indexOf('.');
+    if (dotIndex >= 0) {
+      typeName = typeName.substring(dotIndex + 1);
+    }
+    return Name.anyCamel(typeName).join("iterator").toUpperCamel();
+  }
+
+  private static String lowerFirstLetter(String s) {
+    if (s.length() > 0) {
+      s = Character.toLowerCase(s.charAt(0)) + s.substring(1);
+    }
+    return s;
+  }
+
+  private static String upperFirstLetter(String s) {
+    if (s.length() > 0) {
+      s = Character.toUpperCase(s.charAt(0)) + s.substring(1);
+    }
+    return s;
+  }
+
+  @Override
+  public String getGrpcClientTypeName(Interface service) {
+    // getNickname() thinks the client is a message type and prepends a "*".
+    // substring gets rid of the star.
+    return converter.getTypeName(service).getNickname().substring(1) + "Client";
+  }
+
+  @Override
+  public String getGrpcClientConstructorName(Interface service) {
+    return getGrpcClientTypeName(service).replace(".", ".New");
+  }
+
+  @Override
+  public String getCallOptionsTypeName(Interface service) {
+    return clientNamePrefix(service).join("call").join("options").toUpperCamel();
+  }
+
+  @Override
+  public String getDefaultClientOptionFunctionName(Interface service) {
+    return Name.lowerCamel("default")
+        .join(clientNamePrefix(service))
+        .join("client")
+        .join("options")
+        .toLowerCamel();
+  }
+
+  @Override
+  public String getDefaultCallOptionFunctionName(Interface service) {
+    return Name.lowerCamel("default")
+        .join(clientNamePrefix(service))
+        .join("call")
+        .join("options")
+        .toLowerCamel();
+  }
+
+  @Override
+  public String getClientTypeName(Interface service) {
+    return clientNamePrefix(service).join("client").toUpperCamel();
+  }
+
+  @Override
+  public String getClientConstructorName(Interface service) {
+    return Name.lowerCamel("new").join(clientNamePrefix(service)).join("client").toUpperCamel();
+  }
+
+  @Override
+  public String getPackageName() {
+    // packagePath is in form "cloud.google.com/go/library/apiv1";
+    // we want "library".
+    String[] parts = packagePath.split("/");
+    return parts[parts.length - 2];
+  }
+
+  private Name clientNamePrefix(Interface service) {
+    String name = getReducedServiceName(service);
+    // If there's only one service, or the service name matches the package name, don't prefix with
+    // the service name.
+    if (model.getSymbolTable().getInterfaces().size() == 1 || name.equals(getPackageName())) {
```

Comment:
It occurs to me that going from a cardinality of 1 interface to >1 interfaces in an API would produce a breaking change in the generated code. How set is the Go team on keeping this pattern? BTW it would be *really* cool if you could generate the Go libraries into googleapis/api-client-staging.